packages/web-components/src/pane/pane.ts

```diff
@@ -0,0 +1,260 @@
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * Sets the aria-describedby attribute of the pane.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Toggles the state of the pane (open/closed).
+   * @public
+   */
+  public togglePane(): void {
+    if (!this.open && !this.hasInteracted) {
+      // Only update the hasInteracted flag if the pane is closed and hasn't been interacted with yet
+      this.hasInteracted = true;
+    }
+
+    if (!this.open) {
+      this.previousActiveElement = document.activeElement as HTMLElement;
+    } else if (this.previousActiveElement) {
+      Updates.enqueue(() => this.previousActiveElement?.focus());
+    }
+    this.open = !this.open;
+
+    this.$emit('toggled', { open: this.open, position: this.position, controlSize: this.controlSize });
+    if (this.open) this.handleFocus();
+  }
+
+  /**
+   * Shows the pane.
+   * @public
+   */
+  public openPane(): void {
+    this.previousActiveElement = document.activeElement as HTMLElement;
+    this.open = true;
+    this.$emit('opened', { open: true, position: this.position, controlSize: this.controlSize });
+    if (this.open) this.handleFocus();
+  }
+  /**
+   * Hides the pane.
+   * @public
+   */
+  public closePane(): void {
+    this.open = false;
+    this.$emit('closed', { open: false, position: this.position, controlSize: this.controlSize });
+
+    if (this.previousActiveElement) {
+      Updates.enqueue(() => this.previousActiveElement?.focus());
+    }
+  }
+
+  /**
+   * check if the item is a pane
+   */
+  private dialogRoles = ['dialog', 'complementary'];
+  private isPaneElement = (el: HTMLElement | Pane): boolean => {
+    return el instanceof Pane || (isHTMLElement(el) && this.dialogRoles.includes(el.getAttribute('role') as string));
+  };
+
+  protected setItems(): void {
+    console.log('setItems');
+    const slottedPanes = this.querySelectorAll<Pane>('fluent-pane');
+    console.log('slottedPanes', slottedPanes);
+    this.panes = Array.from(slottedPanes);
+
+    /**
+     * Set the context of the Pane
+     * When the pane is a child component of the PaneSwitcher
+     * we want to signal this through an attribute to update styles.
+     */
+    const filteredPaneItems = this.panes?.filter(this.isPaneElement);
+
+    filteredPaneItems?.forEach((item: HTMLElement, index: number) => {
+      console.log(item);
+      if (item instanceof Pane) {
+        item.setAttribute('data-context', `pane-switcher`);
+      }
+    });
+  }
+
+  /**
+   * Handles keydown events for the pane.
+   * @param event - The keyboard event.
+   * @public
+   */
+  public handleKeyDown = (event: KeyboardEvent): void => {
+    if (this.trapFocus && this.open && event.key === keyTab) {
```

Comment:
FAST has a dep for managing trapping focus in these scenarios, is there a reason to not reuse that since it's likely already in a bundle?